skimage/_shared/fused_numerics.pxd

```diff
@@ -0,0 +1,34 @@
+cimport numpy as cnp
+import numpy as np
+
+ctypedef fused np_ints:
+    cnp.int8_t
+    cnp.int16_t
+    cnp.int32_t
+    cnp.int64_t
+
+ctypedef fused np_uints:
+    cnp.uint8_t
+    cnp.uint16_t
+    cnp.uint32_t
+    cnp.uint64_t
+
+ctypedef fused np_anyint:
+    np_uints
+    np_ints
+
+ctypedef fused np_floats:
+    cnp.float32_t
+    cnp.float64_t
+
+ctypedef fused np_complexes:
+    cnp.complex64_t
+    cnp.complex128_t
+
+ctypedef fused np_real_numeric:
+    np_anyint
+    np_floats
+
+ctypedef fused np_numeric:
+    np_real_numeric
+    np_complexes
```

Comment:
I love this file, and have wanted it for a long time. Thank you!!!

Reply:
This should really be part of cython....

Reply:
Also, if you want this, you should take it out of this PR since I'm not sure where the performance drop is coming from here. Cython casting?

Reply:
I agree about Cython building this in. Perhaps that should be your next PR? ;) In the meantime I'm happy to wait for you to hunt down the errant cast... And I'm also happy to merge, since ~90% of our warping is probably float-based.

Reply:
>Perhaps that should be your next PR? ;
I'd also suggest writing a book on Practical Cython 😅. Seems like @hmaarrfk is a rare expert in this matter.